urls.go

```diff
@@ -201,6 +201,19 @@ func normalizeEscape(s string) (string, error) {
 	return escape(u), nil
 }
 
+// parserest parase the content after the first symbol ':'.
+// If there is a port after ':',  ("", url) is returned. e.g. "www.abc.com:80/"
+// (url[:i], url[i+1:]) is returned in other stations.
+func parseRest(url string, i int) (scheme, path string) {
+	portReg := regexp.MustCompile(`^:(\d+)(/|$)`)
```

Comment:
Let's avoid declaring a regexp here; otherwise we would re-compile the regexp every time this function is called.

Can you modify the `portRegexp` variable declared above and use that here?